app/wikipedia.py

```diff
@@ -20,4 +20,27 @@ def checkArticlesExists(pathList):
 						article["title"] = normalised["from"]
 		for article in queryJson["query"]["pages"]:
 			articleInfo[article["title"]] = not "missing" in article
-	return articleInfo
\ No newline at end of file
+	return articleInfo
+
+def checkValidPath(pathList):
+	pathInfo = {}
+	for index in range(len(pathList) - 1):
```

Comment:
Although you are now using pltitles, this does not remedy the initial problem of calling the api once for each article in the list.